omnibus/files/private-chef-cookbooks/private-chef/libraries/preflight_postgres_validator.rb

```diff
@@ -184,15 +186,21 @@ def backend_verify_database_access(connection)
   def backend_verify_postgres_version(connection)
     # Make sure the server is a supported version.
     r = connection.exec('SHOW server_version;')
-    v = r[0]['server_version']
-    major, minor = v.split('.').map(&:to_i)
+    v = r[0]['server_version'].to_f
 
     # Note that we're looking for the same major, and using our minor as the minimum version
     # This provides compatibility with external databases that use < 9.6 before we make use
     # of any features available in > 9.2.
-    unless (major == REQUIRED_MAJOR) && (minor >= REQUIRED_MINOR)
-      fail_with err_CSPG014_bad_postgres_version(v)
+
+    case
+    when v == REQUIRED_VERSION || v == SUPPORTED_VERSION
```

Comment:
I think it would be better to convert the versions to something like `Gem::Version` for comparison, instead of comparing floats.

Reply:
https://github.com/chef/chef-server/blob/master/omnibus/files/private-chef-cookbooks/private-chef/resources/pg_upgrade.rb#L89 might be a good ref.

Reply:
We'd probably want to convert it above. In general Gem version is going to give you much more reliable version comparison and avoid places where 13 looks smaller than 9.

Reply:
Done.